Retroarch 1.22.0 fails to switch core when loading content not intended for the core running in memory

### Is there an existing issue for this?

- [x] This is a bug in RetroArch frontend
- [x] I have searched the existing issues

### Description

The present bug report is linked to the issue [https://github.com/libretro/RetroArch/issues/17847](url) that was marked as completed in [#17920](https://github.com/libretro/RetroArch/pull/17920), but in v1.22.0 it introduced some new unwanted behavior.

Summary of the initial issue: Retroarch v1.21.0 filters out extensions not owned by the running core when browsing inside of zip files. This issue was introduced in v1.21.0.

New (wrong) behavior: when laoding content "X" running in core "A", then without closing the content, loading content "Y" that refers to core "B", Retroarch shows the extensions even if they do not pertain to the current core, but issues just a black screen or a "Failed to load content" error (see relevant log below in the report). Loading the content "X" or "Y" independently (without switching from a core to the other), works correctly

### Expected behavior

When switching from one core to another, the content should load fine
It sould be possible to load content from any core, regardless of the fact the core itself is the one running in memory (**run content with any core, call the interface and load content from another core, retroarch should close current core, open the appropriate core, load the requested content**)

### Steps to reproduce the bug

1. load any core (Main menu, load core)
2. from main menu, select "load content"
3. select a compressed or non compressed content with an extension not handled by the current core
3a. Retroarch will show the items inside of the compressed file if you selected an archive
4. retroarch issues a black screen or "Failed to load content"

Loading the content "X" or "Y" independently (without switching from a core to the other), works correctly

### Version/Commit

1.22.0

### Bisect Results

1.20.0 used to work fine. In 1.21.0 the extensions pertaining to a core not in memory would now show, in 1.22.0 the extensions are visible again, but the content fails to load

### Present in the nightly version

I don't know

### Platform & operating system

Ubuntu 24.04.2 LTS (Linux v6.8)

### Affected Cores

all

### Environment information

[libretro ERROR] ERROR: Unrecognized ROM file type
ERROR: Unrecognized ROM file type
[ERROR] [Content] Failed to load content.
[INFO] [Core] Unloading core...
[INFO] [Core] Unloading core symbols...


### Relevant log output

```shell

```

--- a/intl/msg_hash_us.h
+++ b/intl/msg_hash_us.h
@@ -5151,7 +5151,7 @@ MSG_HASH(
    )
 MSG_HASH(
    MENU_ENUM_SUBLABEL_CORE_SUGGEST_ALWAYS,
-   "Suggest available cores even when a core is already loaded."
+   "Suggest available cores even when a core is manually loaded."
    )
 MSG_HASH(
    MENU_ENUM_LABEL_VALUE_USE_BUILTIN_PLAYER,

--- a/menu/cbs/menu_cbs_ok.c
+++ b/menu/cbs/menu_cbs_ok.c
@@ -1948,8 +1948,8 @@ static bool menu_content_find_first_core(
             &supported);
 
 #ifdef HAVE_DYNAMIC
-   /* Don't suggest cores if a core is already loaded. */
-   if (     !path_is_empty(RARCH_PATH_CORE)
+   /* Don't suggest cores if a core is manually loaded. */
+   if (     !path_is_empty(RARCH_PATH_CORE_LAST)
          && !config_get_ptr()->bools.core_suggest_always)
       load_content_with_current_core = true;
 #endif

--- a/menu/menu_driver.c
+++ b/menu/menu_driver.c
@@ -8114,6 +8114,14 @@ int generic_menu_entry_action(
       menu_st->flags &= ~(MENU_ST_FLAG_PENDING_CLOSE_CONTENT
                         | MENU_ST_FLAG_PENDING_ENV_SHUTDOWN_FLUSH);
       menu_st->pending_env_shutdown_content_path[0] = '\0';
+
+      /* Reload core on launch failure if manually loaded */
+      if (     !path_is_empty(RARCH_PATH_CORE_LAST)
+            && !(menu_st->flags & MENU_ST_FLAG_PENDING_RELOAD_CORE))
+      {
+         menu_st->flags |= MENU_ST_FLAG_PENDING_RELOAD_CORE;
+         menu_st->flags |= MENU_ST_FLAG_PENDING_ENV_SHUTDOWN_FLUSH;
+      }
    }
    else if (menu_st->flags & MENU_ST_FLAG_PENDING_RELOAD_CORE)
    {

--- a/runloop.c
+++ b/runloop.c
@@ -4291,6 +4291,10 @@ static bool event_init_content(
       /* Single-click playlist return */
       if (settings->bools.input_menu_singleclick_playlists)
          menu_state_get_ptr()->flags |= MENU_ST_FLAG_PENDING_CLOSE_CONTENT;
+
+      /* Return from empty Quick Menu if core is manually loaded and needs reloading */
+      if (!path_is_empty(RARCH_PATH_CORE_LAST))
+         menu_state_get_ptr()->flags |= MENU_ST_FLAG_PENDING_CLOSE_CONTENT;
 #endif
       return false;
    }

